### ISSUE TITLE
Illuminate\Database\QueryException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '12345678@students.lincoln....

### ISSUE THREAD

**Issue by marc-hanheide** [LCAS/pgr_system#56]
https://sentry.io/university-of-lincoln/pgr-system/issues/672494030/

```
PDOException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '12345678@students.lincoln.ac.uk' for key 'users_university_email_unique'
  File "app/Http/Controllers/StudentController.php", line 195, in store
    'user_type' => 'Student',
  File "app/Http/Middleware/SamlAuth.php", line 26, in handle
    return $next($request);
  File "app/Http/Middleware/DevUser.php", line 25, in handle
    return $next($request);
  File "app/Http/Middleware/RoutePrefixProxy.php", line 25, in handle
    return $next($request);
  File "public/index.php", line 52
    $request = Illuminate\Http\Request::capture()
...
(73 additional frame(s) were not displayed)

Illuminate\Database\QueryException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '12345678@students.lincoln.ac.uk' for key 'users_university_email_unique' (SQL: insert into `users` (`first_name`, `last_name`, `university_id`, `university_email`, `user_type`, `updated_at`, `created_at`) values (Sam, Jones, 12345678, 12345678@students.lincoln.ac.uk, Student, 2018-09-03 09:21:28, 2018-09-03 09:21:28))
  File "app/Http/Controllers/StudentController.php", line 195, in store
    'user_type' => 'Student',
  File "app/Http/Middleware/SamlAuth.php", line 26, in handle
    return $next($request);
  File "app/Http/Middleware/DevUser.php", line 25, in handle
    return $next($request);
  File "app/Http/Middleware/RoutePrefixProxy.php", line 25, in handle
    return $next($request);
  File "public/index.php", line 52
    $request = Illuminate\Http\Request::capture()
...
(71 additional frame(s) were not displayed)
```

**Comment by pet1330** [comment]
Fist or create matches all elements, so we need to validate if only the university ID matches first. 

https://github.com/LCAS/pgr_system/blob/5be2937f47e9610105a6c2e0051db11de9179bb7/app/Http/Controllers/StudentController.php#L155

I'll fix this later today so it cannot happen again

**Comment by pet1330** [comment]
I've had a play and although the above would allow a user (in theory) to put a second student into the database, I am unable to reach this function through the web interface with a student that is already in the database. If you can write a test to replicate this error then I will look for a fix.

**Comment by pet1330** [comment]
This can occur when an Admins miss spell a users name, as it tries to match all of the attributes when finding that user, rather than just their university email. This has now been fixed in: 95453aa
